### PR TITLE
net/local:set POLLIN/POLLOUT threshold for local fifo

### PR DESCRIPTION
**Pull request by XinStellaris**
## Summary
This is the next step of https://github.com/apache/nuttx/pull/8052
When local udp socket is used, this threshold will avoid non-blocking read failed with -EAGAIN when only preamble len is sent and read by reader. To do so, local udp socket should set threshold to be sizeof preamble len.

## Impact
Local udp socket only

## Testing
Tested on ESP32C3 after removing some #warning macro in local udp socket

### CODE REVIEW

**Comment by XinStellaris** [comment]
The local udp socket has some #warning , for example:

```
#ifdef CONFIG_NET_LOCAL_DGRAM
int local_release_halfduplex(FAR struct local_conn_s *conn)
{
#if 1
  /* REVISIT: We need to think about this carefully.  Unlike the connection-
   * oriented Unix domain socket, we don't really know the best time to
   * release the FIFO resource.  It would be extremely inefficient to create
   * and destroy the FIFO on each packet. But, on the other hand, failing
   * to destroy the FIFO will leave the FIFO resources in place after the
   * communications have completed.
   *
   * I am thinking that there should be something like a timer.  The timer
   * would be started at the completion of each transfer and cancelled at
   * the beginning of each transfer.  If the timer expires, then the FIFO
   * would be destroyed.
   */

#  warning Missing logic
  return OK;

#else
  char path[LOCAL_FULLPATH_LEN];

  /* Destroy the half duplex FIFO if it exists. */

  local_hd_name(conn->lc_path, path);
  return local_release_fifo(path);
#endif
}
#endif /* CONFIG_NET_LOCAL_DGRAM */
```
 I guess no one used it before? Anyway, I fixed the potential bug.

File: net/local/local_fifo.c [hunk 1]
```diff
@@ -603,6 +657,17 @@ int local_open_receiver(FAR struct local_conn_s *conn, bool nonblock)
       /* Policy: Free FIFO resources when the buffer is empty. */
 
       ret = local_set_policy(&conn->lc_infile, 1);
+
+      if (ret == 0)
```
Comment (xiaoxiang781216) [inline net/local/local_fifo.c:661]: only nonblock need set threshold?

File: net/local/local_fifo.c [hunk 2]
```diff
@@ -636,6 +701,16 @@ int local_open_sender(FAR struct local_conn_s *conn, FAR const char *path,
       /* Policy: Free FIFO resources when the buffer is empty. */
 
       ret = local_set_policy(&conn->lc_outfile, 1);
+      if (ret == 0)
```
Comment (xiaoxiang781216) [inline net/local/local_fifo.c:704]: ditto

File: net/local/local_fifo.c (hunk 1 above)
Comment (XinStellaris) [reply net/local/local_fifo.c:661]: The key thought is:
POLLIN/ POLLOUT shouldn't happen if only preamble can be buffered, whether block or nonblock.

Though only nonblock will lead to a much worse result: packet receiving failure. I don't think we should distinguish between block and non-block.

Comment (xiaoxiang781216) [reply net/local/local_fifo.c:661]: Ok.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**